zookeeper-server/src/main/java/org/apache/zookeeper/server/quorum/Learner.java

```diff
@@ -116,10 +120,14 @@ public Socket getSocket() {
 
     public static final String LEARNER_ASYNC_SENDING = "learner.asyncSending";
     private static boolean asyncSending = Boolean.getBoolean(LEARNER_ASYNC_SENDING);
+    public static final String LEARNER_CLOSE_SOCKET_ASYNC = "learner.closeSocketAsync";
+    public static final boolean closeSocketAsync = Boolean.getBoolean(LEARNER_CLOSE_SOCKET_ASYNC);
```

Comment:
Not trying to block this PR, and things can be improved later, but:

I see that such properties, including `learner.asyncSending` above it, are not documented.  Is there a specific policy regarding which properties are to be mentioned in `zookeeperAdmin.md`?  I am wondering how one can learn about all these knobs, and their raison d'être?  (I know the corresponding ticket ID can be found via Git, but that incurs some overhead.)

Reply:
Thanks for your comments. I will add the flag to zookeeperAdmin. As for test, I find it is hard to construct a unit test for this. But this feature has been enabled in our production for months.

Reply:
@jhuan31: By "untested," I meant: I have looked into the code, but *I* haven't tested it :)  I understand that it is not always easy to create meaningful tests for socket behavior.  Thank you for taking care of the documentation.